src/pages/index.tsx

```diff
@@ -1,22 +1,68 @@
 import { EventCarousel } from '@/components/events';
+import Hero from '@/components/home/Hero';
+import { showToast } from '@/lib';
 import { EventAPI } from '@/lib/api';
 import withAccessType from '@/lib/hoc/withAccessType';
+import { attendEvent } from '@/lib/managers/EventManager';
 import { CookieService, PermissionService } from '@/lib/services';
-import type { PublicAttendance, PublicEvent } from '@/lib/types/apiResponses';
+import type {
+  CustomErrorBody,
+  PrivateProfile,
+  PublicAttendance,
+  PublicEvent,
+} from '@/lib/types/apiResponses';
 import { CookieType } from '@/lib/types/enums';
 import styles from '@/styles/pages/Home.module.scss';
 import { GetServerSideProps } from 'next';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
 
 interface HomePageProps {
+  user: PrivateProfile;
+  authToken: string;
   pastEvents: PublicEvent[];
   upcomingEvents: PublicEvent[];
   liveEvents: PublicEvent[];
   attendances: PublicAttendance[];
 }
 
-const PortalHomePage = ({ pastEvents, upcomingEvents, liveEvents, attendances }: HomePageProps) => {
+const checkin = async (token: string, attendanceCode: string): Promise<void> => {
+  const onSuccessCallback = (event: PublicEvent) => {
+    const title = `Checked in to ${event.title}!`;
+    const subtitle = `Thanks for checking in! You earned ${event.pointValue} points.`;
+    showToast(title, subtitle);
+  };
+  const onFailCallback = (error: CustomErrorBody) => {
+    showToast('Unable to checkin!', error.message);
+  };
+  await attendEvent({ token, attendanceCode, onSuccessCallback, onFailCallback });
+};
+
+const PortalHomePage = ({
+  user,
+  authToken,
+  pastEvents,
+  upcomingEvents,
+  liveEvents,
+  attendances,
+}: HomePageProps) => {
+  const router = useRouter();
+
+  useEffect(() => {
+    if (typeof router.query.code === 'string' && router.query.code) {
+      // In dev mode, this runs twice because of reactStrictMode in nextConfig.
+      // This will only be run once in prod or deployment.
+      checkin(authToken, router.query.code).then(() => {
+        // Clear out the query params without re-running getServerSideProps.
+        router.replace('/', undefined, { shallow: true });
+      });
+    }
+  }, [authToken, router]);
```

Comment:
do you think this could be done on the server side, in `getServerSidePropsFunc`?